lib/ansible/modules/cloud/openstack/os_tempest_run.py

```diff
@@ -0,0 +1,296 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: os_tempest_run
+short_description: Runs Tempest
+description:
+    - Runs Tempest according to the configuration file in the given workspace
+
+version_added: "2.4"
+
+author: "Tal Shafir , @TalShafir"
+requirements: ["Tempest"]
+options:
+    workspace:
+        description:
+            The workspace as was configured in 'Tempest init <I(workspace)>'
+        required: True
+    dest:
+        description:
+            Path for the output from Tempest, if not given the result will be printed in the exit json under 'out' and 'err'
+        required: True
+    regex:
+        description:
+            Selection regex for tests, Tempest will run any tests that match on the I(regex)
+            must use format for python's re.match()
+        required: False
+        default: ""
+    whitelist_file:
+        description:
+            - Path for a file with a line separated list of regex, Tempest will run only tests that match at least one regex.
+            - Cannot work with the I(blacklist_file) argument
+        required: False
+        default: ""
+    blacklist_file:
+        description:
+            - Path for a file with a line separated list of regex, Tempest won't run tests that match at least one regex.
+            - Cannot work with the I(whitelist_file) argument
+        required: False
+        default: ""
+    concurrency:
+        description:
+            Number of workers to use(the default is one worker for each CPU core), set to C(1) in case you want to run serially
+        required: False
+        default: None
+    force:
+        description:
+            When C(True) override output file if already exists
+        required: False
+        default: 'False'
+    subunit:
+        description:
+            When C(True) show the output as subunit v2
+        required: False
+        default: 'False'
+notes:
+    - You can find out more about Tempest at U(http://docs.openstack.org/developer/tempest/)
+    - The module requires to tempest init <I(workspace)> before usage
+    - The options I(whitelist_file) and I(blacklist_file) are mutually exclusive, if they are both given only the I(whitelist_file) will be used
+'''
+
+EXAMPLES = '''
+# Run all tests with default number of workers
```

Comment:
That's a good idea.

Where in the existing documentation/examples did you look?

I'm wondering where I can put/link to the example so people will find it.